neural_net_basics/neural-network.md

```diff
@@ -0,0 +1,137 @@
+
+### Logistic Function a.k.a Sigmoid function
+![png](/neural_net_basics//assets//sigmoid.png)
+$$ f(x) = \frac{1}{1 + e^{-x}}$$
+The Logistic function, a variant of the sigmoid function, initially replaced the step function. Like the Step function it limits the output of a neuron between 0 and 1, but doesn't have the same issues around the zero mark. This makes it easier to train as a change to the weights and biases will always have an effect, except at the extremes.
+
+### Rectified Linear Unit (ReLU)
+![png](/neural_net_basics//assets//relu.png)
+$$ f(x) = max(0,x) $$
+The Rectified Linear  (ReLU) has become the most popular and in most cases the default activation function to use for deep neural networks. It takes the maximum value of either 0 or the input, essentially putting any negative input value to 0 immediately.
+
+### SoftMax
+$$ \delta(z) = \frac{e^{z_{i}}}{\sum_{j=1}^{K}e^{z_{j}}} $$
+The SoftMax Activation function is used for the final output layer for classification neural networks. It converts the output values into a probability distribution over the output classes. Each output neuron will represent the probability the input is of that class.
+
+## Architecture of a Neural Network
+
+In neural networks, the neurons can be arranged in a variety of ways. For the basic neural network the neurons are arranged into several distinct layers. 
+
+There are three layer categories
+
+1. Input Layer
+2. Hidden Layers
+3. Output Layer
+
+### 1. Input Layer
+The input layer is purely the values of the input data, no calculations take place. There is a neuron for each input value, for image classification neural networks, for grey scale images, there would be a node for each pixel. 
+
+### 2. Hidden Layers
+If there is a single hidden layer then the neural network is considered to be a shallow neural network. If there are two or more hidden layers then it is considered to be a deep learning neural network.
+
+There can be any number of neurons in each hidden layer.
+
+### 3. Output Layer
+The output layer gives the output of the neural network. For image classification there would be a node for each class of item to be classified, for example in hand written digit recognition there would be 10 nodes, one for each of the 10 digits.
+
+### How the layers are connected
+
+For the basic layered neural network, each neuron takes the outputs of every neuron in the previous layer as it’s input. With the first hidden layer taking the inputs to the neural network directly.
+
+![png](/neural_net_basics//assets//architecture.png)
+
+For a classifier neural network, the output layer has neuron that relates to specific classes of output. For example in a neural network that classifies numerical digits, of which there are 10, you would have 10 output neurons. The neuron with the highest output signifies which class the network has identified. 
```

Comment:
This is the same example used a couple of paragraphs up.

Reply:
I've removed it and simplified this section